### DISCUSSION TITLE
Multiple trigger elements to open menu in Autocomplete?

### DISCUSSION THREAD

**Discussion by ernes-to** [floating-ui/floating-ui#1647] [Q&A] [answered]
Hi there. Here's a [Codesandbox](https://codesandbox.io/s/floatingui-autocomplete-shared-t6kxcv?file=/src/App.tsx:4381-4397) using FloatingUI forked from the React Autocomplete [example](https://floating-ui.com/docs/react-dom-interactions#select-listbox) in the docs. 

I've added a search icon that sits on top of the text input, and I'd like to toggle the menu when the user clicks on it. Currently, clicking on this button to open it works as expected, but clicking it again to close it causes it close and immediately open again.

I've read several parts of the docs and tried a few different things that haven't worked out, so I'm wondering what the best way is to have multiple trigger elements that open/close the menu (and maintain proper keyboard navigation context).

Thanks!

**Comment by atomiks** [comment, marked as answer]
In a combobox I believe this is called a "toggle button" and shouldn't be tabbable. The ARIA 1.1 page shows it as "Example 3: List with Inline Autocomplete" although they have the toggle issue you describe.

For keydowning to work, either the reference or floating element needs focus. So you can move focus back to the input when you click it, and exclude it from the tab order. The `onBlur` also needs tweaking:
https://codesandbox.io/s/floatingui-autocomplete-shared-forked-k90mu7?file=/src/App.tsx

**Reply by ernes-to** [reply]
@atomiks Thank you for your guidance on this! What you did in the code made sense, and yeah it is interesting that the example on the ARIA 1.1 page also exhibits this behavior. You were also right in thinking that something else was going to be needed to get this to work as expected in Safari. As it turns out, there's this [bug](https://bugs.webkit.org/show_bug.cgi?id=22261) on WebKit (which looks like it won't be fixed) reported nearly 14 years ago about click events not triggering the focus event on buttons and [other elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#clicking_and_focus).

Aside from a polyfill, or triggering focus manually via refs + adding additional logic to the `onBlur` event in the input, I opted for switching the `button` to be a `div` with a `role="button"` which does receive focus properly on click across all browsers. I'm curious to know if you think there's a better solution than that though. Regardless, thanks again 👍
